### PR TITLE
add regex to get componentID from control mapping

### PR DESCRIPTION
**Pull request by aitchkhan**
add componentID from excel  in generated implementation

### CODE REVIEW

File: impl/impl_test.go [hunk 1]
```diff
@@ -29,7 +29,7 @@ func TestGenerateImplementation(t *testing.T) {
 	}
 	for i, x := range ComponentDetails {
 		csvs[i+rowIndex][controlIndex] = x[0]
-		csvs[i+rowIndex][componentNameIndex] = x[1]
```
Comment (minhaj10p) [inline impl/impl_test.go]: if we're taking names seriously, lets make this `componentConf`

File: impl/implementation.go [hunk 2]
```diff
@@ -12,17 +12,20 @@ import (
 const (
 	//totalControlsInExcel the total number of controls in the excel sheet
 	totalControlsInExcel = 264
-	//componentNameIndex The Column at which name of the component configuration is present
-	componentNameIndex = 16
```
Comment (minhaj10p) [inline impl/implementation.go]: same here

File: impl/implementation.go [hunk 3]
```diff
@@ -12,17 +12,20 @@ import (
...
 	//controlIndex Column at which control is present in the excel sheet
 	controlIndex = 2
 	//rowIndex Starting point for valid rows (neglects titles)
-	rowIndex  = 3
-	delimiter = "|"
+	rowIndex = 3
+	// componentNameRow is the index for getting component name
+	componentNameRow = 1
+	delimiter        = "|"
+	componentIDRegex = "cpe:[0-9].[0-9]:[a-z]:docker:[a-z]*:[0-9].[0-9].[0-9]"
```
Comment (minhaj10p) [inline impl/implementation.go]: (Y)

**Comment by aitchkhan** [comment]
which slice indices are you referring to? 
the one generating the [componentDefiniton ](https://github.com/docker/oscalkit/pull/51/files#diff-09d9a02be4dc559059e2aa5d8dfa0da6R194) or the one consuming the index of excel sheet for extracting [identifier](https://github.com/docker/oscalkit/pull/51/files#diff-09d9a02be4dc559059e2aa5d8dfa0da6R194) of UCP?

**Comment by asadullah-yousuf-10p** [comment]
> which slice indices are you referring to?
> the one generating the [componentDefiniton ](https://github.com/docker/oscalkit/pull/51/files#diff-09d9a02be4dc559059e2aa5d8dfa0da6R194) or the one consuming the index of excel sheet for extracting [identifier](https://github.com/docker/oscalkit/pull/51/files#diff-09d9a02be4dc559059e2aa5d8dfa0da6R194) of UCP?

@anweiss

**Comment by anweiss** [comment]
@haroonKhan-10p if you could fix the iteration constructs by removing the hardcoded indices, we can go ahead and get this merged. thanks

**Comment by aitchkhan** [comment]
@anweiss the componentdefinition iteration index?
Because the CSV indices used are identifier for different things like
uuids, narratives.

**Comment by anweiss** [comment]
@mohuk @aitchkhan we can leave the hardcoded indices for now. Please rebase and resolve conflicts and I'll go ahead and merge. I'll create a separate issue for us to address this in the future when the OSCAL "implementation" model matures.

**Comment by anweiss** [comment]
@aitchkhan @haroonKhan-10p please resolve conflicts and rebase.

**Comment by haroonKhan-10p** [comment]
updated.

**Pull request merged**